src/main/java/duke/main/Storage.java

```diff
@@ -0,0 +1,66 @@
+package duke.main;
+
+import duke.task.*;
+import duke.exception.DukeException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.FileWriter;
+
+public class Storage {
+    private String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public ArrayList<Task> load() throws DukeException {
+        try {
+            ArrayList<Task> tasks = new ArrayList<Task>();
+            Scanner sc = new Scanner(new File(filePath));
+            while (sc.hasNextLine()) {
+                String[] task = sc.nextLine().split(" \\| ");
+                switch (task[0]) {
+                case "[T]":
+                    tasks.add(new Todo(task[2], (task[1].equals("[\u2713]") ? true : false)));
+                    break;
+                case "[D]":
+                    tasks.add(new Deadline(task[2], task[3].substring(4), (task[1].equals("[\u2713]") ? true : false)));
+                    break;
+                case "[E]":
+                    tasks.add(new Event(task[2], task[3].substring(4), (task[1].equals("[\u2713]") ? true : false)));
+                    break;
+                }
+            }
+            return tasks;
+        }
+        catch (Exception e) {
+            throw new DukeException("OOPS!!! " + e.getMessage());
```

Comment:
Acceptable to just throw IOException from the reading of the file instead of converting from one form of exception to another